util_test.go

```diff
@@ -33,3 +57,166 @@ func TestFilterCustomQuery(t *testing.T) {
 		t.Fatalf("expected ok=y, got %v", nx.Get("ok"))
 	}
 }
+
+// TestHasSQLMigration verifies detection of executable SQL vs
+// comments/whitespace. Data is considered to be executable SQL, if the data
+// contains anything else than just comments/whitespace.
+func TestHasSQLMigration(t *testing.T) {
+	tests := []struct {
+		name         string
+		data         string
+		expectHasSQL bool
+	}{
+		{
+			name:         "empty string",
+			data:         "",
+			expectHasSQL: false,
+		},
+		{
+			name:         "whitespace and semicolons only",
+			data:         "\n;\t;\r\n  ;",
+			expectHasSQL: false,
+		},
+		{
+			name:         "only comments",
+			data:         "-- comment line\n/* block comment */\n;",
+			expectHasSQL: false,
+		},
+		{
+			name:         "bom with no statements",
```

Comment:
what is the meaning of "bom" ?

Reply:
It stands for the `\uFEFF` unicode character.
Some editors insert it automatically at the very beginning of a file, which may appear invisibly to indicate:
* The text is encoded in UTF-8, UTF-16, or UTF-32
* The byte order (endianness) for UTF-16/UTF-32